plugins/inputs/jti_openconfig_telemetry/openconfig_telemetry.go

```diff
@@ -0,0 +1,324 @@
+		if loginErr != nil {
+			log.Fatalf("E! Could not initiate login check: %v", err)
+		}
+
+		// Check if the user is authenticated. Bail if auth error
+		if !loginReply.Result {
+			log.Fatalf("E! Failed to authenticate the user")
+		}
+	}
+
+	c := telemetry.NewOpenConfigTelemetryClient(m.grpcClientConn)
+
+	for _, sensor := range m.Sensors {
+		go func(sensor string, acc telegraf.Accumulator) {
+			spathSplit := strings.SplitN(sensor, " ", -1)
+			var sensorName string
+			var pathlist []*telemetry.Path
+			customReportingRate := false
+			var reportingRate uint64
+
+			reportingRate = uint64(m.SampleFrequency)
+
+			if len(spathSplit) > 1 {
+				for i, path := range spathSplit {
+					// We allow custom reporting rate per sensor by specifying rate in milliseconds
+					// followed by measurement name or sensor path
+					if i == 0 {
+						sensorName = path
+						reportingRate, err = strconv.ParseUint(path, 10, 32)
+						if err == nil {
+							customReportingRate = true
+						} else {
+							reportingRate = uint64(m.SampleFrequency)
+							// If the first word is not an integer and starts with /, we treat it as both sensor and a measurement name
+							if strings.HasPrefix(sensorName, "/") {
+								pathlist = append(pathlist, &telemetry.Path{Path: path, SampleFrequency: uint32(reportingRate)})
+							}
+						}
+					} else if customReportingRate && i == 1 {
+						// If our first word is reporting rate for this list of sensors, we treat second word as measurement name and if it has /
+						// we treat it as another sensor to be monitored
+						sensorName = path
+						if strings.HasPrefix(path, "/") {
+							pathlist = append(pathlist, &telemetry.Path{Path: path, SampleFrequency: uint32(reportingRate)})
+						}
+					} else {
+						pathlist = append(pathlist, &telemetry.Path{Path: path, SampleFrequency: uint32(reportingRate)})
+					}
+				}
+			} else {
+				sensorName = sensor
+				pathlist = append(pathlist, &telemetry.Path{Path: sensor, SampleFrequency: uint32(reportingRate)})
+			}
+			stream, err := c.TelemetrySubscribe(context.Background(),
+				&telemetry.SubscriptionRequest{PathList: pathlist})
+			if err != nil {
+				log.Fatalf("E! Could not subscribe: %v", err)
+			}
+			for {
+				r, err := stream.Recv()
+				if err == io.EOF {
+					break
+				}
+				if err != nil {
+					log.Fatalf("E! Failed to read: %v", err)
+				}
+
+				// Print incoming data as info if debug is set
+				if m.Debug {
+					log.Printf("I! Received: %v", r)
+				}
+
+				// Create a point and add to batch
+				tags := make(map[string]string)
+
+				if err != nil {
+					log.Fatalln("E! Error: %v", err)
+				}
+
+				// Use empty prefix. We will update this when we iterate over key-value pairs
+				prefix := ""
+
+				// Insert additional tags
+				tags["device"] = grpc_server
+
+				dgroups := []DataGroup{}
+
+				for _, v := range r.Kv {
```

Comment:
Can you provide some sample data from an actual device so that I can better evaluate the data schema?

Reply:
system_id:jtidevice
component_id:65535
path:sensor_1000:/interfaces/interface/state/oper-status/:/interfaces/interface/state/oper-status/:mib2d
sequence_number:0
timestamp:1513886669462
kv {
  key:__timestamp__
  uint_value:1513886671466
}
kv {
  key:__junos_re_stream_creation_timestamp__
  uint_value:1513886669430
}
kv {
  key:__junos_re_payload_get_timestamp__
  uint_value:1513886669430
}
kv {
  key:__prefix__
  str_value:/interfaces/interface[name='lsi']/
}
kv {
  key:state/oper-status
  str_value:UP
}